tests/unit/lib/test_opensearch_tls.py

```diff
@@ -1578,7 +1578,11 @@ def test_on_certificate_available_rotation_ongoing_on_this_unit(
         # No action taken, no change on status or certificates
         assert run_cmd.call_count == 0
         assert self.harness.model.unit.status == original_status
-        self.charm.on.certificate_available.defer.called_once()
+        if leader:
```

Comment:
Seems we are mixing `leader=True` and `leader=False`, and defer will be called only for leader units. We should not limit deferrals to leaders only. That will cause issues if we lose the leader unit later.